library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -129,7 +129,11 @@ private DownloadBatchStatusCallback downloadBatchCallback(Map<DownloadBatchId, D
                     for (DownloadBatchStatusCallback callback : callbacks) {
                         callback.onUpdate(downloadBatchStatus);
                     }
-                    notificationDispatcher.updateNotification(downloadBatchStatus);
+
+                    DownloadBatch downloadBatch = downloadBatchMap.get(downloadBatchId);
+                    if (downloadBatch != null) {
```

Comment:
We can't use the line 115's `downloadBatchStatus` as it is a copy of the underlying status. In order to update the `notificationSeen` we need the one that is modified and persisted by the `download-manager`, so we pull it from our source of truth which is our map. 

This knowledge about `copied` vs `real` download batch status is a bit hidden, we should spend sometime updating the `DownloadBatchStatusCallback` to make this relationship more explicit.